### PR TITLE
Adding unlimited pagination 

### PR DESCRIPTION
**Pull request by sdsomma**
Pagination can now be ignored by setting the search request to unlimited.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -137,6 +137,17 @@ $databaseQuery->take($limit)->skip(($page - 1) * $limit);
...
+If you want to ignore the pagination for a request, you can call the `unlimited()` or `all()` methods.
+
+```php
+$request = $request->unlimited();
+
+if (!$request->isUnlimited())
+   $databaseQuery->take($request->getLimit())->skip($request->getSkip());
+```
+
+The unlimited feature is only a falg. Your database query applier must check for this with the `isUnlimited()` method to support this feature.
```
Comment (janhartigan) [inline README.md]: falg

**Review by janhartigan — CHANGES_REQUESTED** [review]
There should probably be a way to reverse this. What would be good enough, I think, is to always set `unlimited` to false whenever you set the `limit()`

**Comment by janhartigan** [comment]
Also needs js counterpart if you aren't already doing it

**Comment by sdsomma** [comment]
i started making deeply tied to page and limit.. then it got a little weird.. so i decided to make it a flag that unrelated to pagination.. the applier can decide if it wants to use it or not.. else you can potentially get someone from fronted send unlimited on a request that you would not want to allow... you can unset it by ->unlimited(false)

**Comment by sdsomma** [comment]
but yea.. maybe setting a limit or page should mark unlimited as false.. makes sense

**Comment by sdsomma** [comment]
js comming

File: src/SearchRequest.php [hunk 2]
```diff
@@ -122,9 +122,9 @@ public function overrideWithJson($json)
 		$this->addFacets($inputs['facets']);
 		$this->groupBy($inputs['groups']);
 		$this->addFilterSetFromArray($inputs['filterSet']);
-		$this->page = $inputs['page'];
-		$this->limit = $inputs['limit'];
 		$this->unlimited = isset($inputs['unlimited']) ? $inputs['unlimited'] : false;
+		$this->page($inputs['page']);
+		$this->limit($inputs['limit']);
```
Comment (janhartigan) [inline src/SearchRequest.php]: I think this creates a failing test scenario:

```
$originalRequest = SearchRequest::create()->limit(1)->unlimited();
$newRequest = SearchRequest::create($originalRequest->toJson());

$this->assertTrue($newRequest->isUnlimited());
```

**Review by janhartigan — CHANGES_REQUESTED** [review]

**Review by janhartigan — APPROVED** [review]

**Pull request merged**